openedx/features/course_duration_limits/tests/test_models.py

```diff
@@ -0,0 +1,177 @@
+            enabled=True,
+            course=self.course_overview,
+            enabled_as_of=date.today(),
+        )
+
+        if already_enrolled:
+            if enrolled_before_enabled:
+                enrollment_create_date = config.enabled_as_of - timedelta(days=1)
+            else:
+                enrollment_create_date = config.enabled_as_of + timedelta(days=1)
+
+            today = None
+
+            existing_enrollment = CourseEnrollmentFactory.create(
+                user=self.user,
+                course=self.course_overview,
+            )
+            existing_enrollment.created = datetime.combine(enrollment_create_date, time())
+            existing_enrollment.save()
+        else:
+            existing_enrollment = None
+            if enrolled_before_enabled:
+                today = config.enabled_as_of - timedelta(days=1)
+            else:
+                today = config.enabled_as_of + timedelta(days=1)
+
+        if pass_enrollment:
+            enrollment = existing_enrollment
+            user = None
+            course_key = None
+        else:
+            enrollment = None
+            user = self.user
+            course_key = self.course_overview.id
+
+        if pass_enrollment:
+            query_count = 4
+        else:
+            query_count = 5
+
+        with self.assertNumQueries(query_count):
+            enabled = CourseDurationLimitConfig.enabled_for_enrollment(
+                enrollment=enrollment,
+                user=user,
+                course_key=course_key,
+                today=today,
+            )
+            self.assertEqual(not enrolled_before_enabled, enabled)
+
+    def test_enabled_for_enrollment_failure(self):
+        with self.assertRaises(ValueError):
+            CourseDurationLimitConfig.enabled_for_enrollment(None, None, None)
+        with self.assertRaises(ValueError):
+            CourseDurationLimitConfig.enabled_for_enrollment(Mock(name='enrollment'), Mock(name='user'), None)
+        with self.assertRaises(ValueError):
+            CourseDurationLimitConfig.enabled_for_enrollment(Mock(name='enrollment'), None, Mock(name='course_key'))
+
+    @override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+    def test_enabled_for_enrollment_flag_override(self):
+        self.assertTrue(CourseDurationLimitConfig.enabled_for_enrollment(None, None, None))
+        self.assertTrue(CourseDurationLimitConfig.enabled_for_enrollment(Mock(name='enrollment'), Mock(name='user'), None))
+        self.assertTrue(CourseDurationLimitConfig.enabled_for_enrollment(Mock(name='enrollment'), None, Mock(name='course_key')))
+
+    @ddt.data(True, False)
+    def test_enabled_for_course(
+        self,
+        before_enabled,
+    ):
+        config = CourseDurationLimitConfig.objects.create(
+            enabled=True,
+            course=self.course_overview,
+            enabled_as_of=date.today(),
+        )
+
+        if before_enabled:
```

Comment:
this seems like it is handling the enabled_as_of edge case for the same day. could you clarify this case with an example?

Reply:
I'm not sure I understand which edge case you're asking about.

Reply:
i just meant that a verbal example/explanation of why we're doing  +1 day and -1 day would be helpful